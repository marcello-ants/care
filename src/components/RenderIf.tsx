import React, { ReactElement } from 'react';

interface IProps {
  condition: boolean;
  children: React.ReactChild;
}

const RenderIf = ({ condition, children }: IProps): ReactElement => {
  if (condition) {
    return <>{children}</>;
  }
  return <></>;
};

export default RenderIf;
