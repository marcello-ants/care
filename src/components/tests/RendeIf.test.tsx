import { render, screen } from '@testing-library/react';

import RenderIf from '../RenderIf';

describe('RenderIf', () => {
  it('Matches snapshot', () => {
    const { asFragment } = render(
      // eslint-disable-next-line react/jsx-boolean-value
      <RenderIf condition={true}>
        <p>This is a test</p>
      </RenderIf>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('Renders a component when condition is met', () => {
    render(
      // eslint-disable-next-line react/jsx-boolean-value
      <RenderIf condition={true}>
        <p>This is a test</p>
      </RenderIf>
    );

    const textElement = screen.getByText('This is a test');
    expect(textElement).toBeInTheDocument();
  });

  it('Do not render a component when condition is not met', () => {
    render(
      // eslint-disable-next-line react/jsx-boolean-value
      <RenderIf condition={false}>
        <p data-testid="test">This is a test</p>
      </RenderIf>
    );

    const textElement = screen.queryByTestId('test');
    expect(textElement).not.toBeInTheDocument();
  });
});
