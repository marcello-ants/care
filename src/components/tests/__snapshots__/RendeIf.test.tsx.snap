// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderIf Matches snapshot 1`] = `
<DocumentFragment>
  <p>
    This is a test
  </p>
</DocumentFragment>
`;
